builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
+	ul, ok := data.GetOk("username_length")
```

Comment:
We can add the `Default` value in the FieldSchema of `username_length`.
Then, this can be changed to `ul := data.Get("username_length")` and if-else block can be skipped.